### PR TITLE
feature(rolling upgrade): cover all kinds of compaction strategies

### PR DESCRIPTION
**Pull request by amoskong**
Ticket: https://trello.com/c/ug7CamXC/1849-enhance-rolling-upgrade-with-tables-of-all-compaction-strategies

Currently LeveledCompactionStrategy and SizeTieredCompactionStrategy are ONLY
used in one subtest (table_options_test). The default SizeTieredCompactionStrategy
is used for all c-s workloads and other subtests in fill_db_data.py

This patch changed to use all kinds of compaction strategies in upgrade
test.

Backport note: IncrementalCompactionStrategy should be added when this
patch is backported to enterprise.

Signed-off-by: Amos Kong <amos@scylladb.com>

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by amoskong** [comment]
BYO job: https://jenkins.scylladb.com/job/scylla-staging/job/amos/job/rolling-upgrade-centos-clone/8/

- running .....

**Review by juliayakovlev — APPROVED** [review]

File: test-cases/upgrades/rolling-upgrade.yaml [hunk 1]
```diff
@@ -3,12 +3,12 @@
...
-stress_cmd_read_10m: cassandra-stress read no-warmup cl=QUORUM duration=10m -schema 'replication(factor=3) compression=SnappyCompressor' -port jmx=6868 -mode cql3 native compression=snappy -rate threads=1000 -pop seq=1..10100200 -log interval=5
-stress_cmd_read_60m: cassandra-stress read no-warmup cl=QUORUM duration=60m -schema 'replication(factor=3) compression=LZ4Compressor' -port jmx=6868 -mode cql3 native compression=lz4 -rate threads=1000 -pop seq=1..10100200 -log interval=5
+stress_cmd_read_10m: cassandra-stress read no-warmup cl=QUORUM duration=10m -schema 'replication(factor=3) compression=SnappyCompressor compaction(strategy=TimeWindowCompactionStrategy)' -port jmx=6868 -mode cql3 native compression=snappy -rate threads=1000 -pop seq=1..10100200 -log interval=5
+stress_cmd_read_60m: cassandra-stress read no-warmup cl=QUORUM duration=60m -schema 'replication(factor=3) compression=LZ4Compressor compaction(strategy=DateTieredCompactionStrategy)' -port jmx=6868 -mode cql3 native compression=lz4 -rate threads=1000 -pop seq=1..10100200 -log interval=5
```
Comment (roydahan) [inline test-cases/upgrades/rolling-upgrade.yaml:11]: c-s ignore the compaction strategy you supply when using "read" command.
it just uses what you had in write.
please remove from here.

Comment (amoskong) [reply test-cases/upgrades/rolling-upgrade.yaml]: fixed.

**Review by roydahan — APPROVED** [review]

**Comment by roydahan** [comment]
If passed on master, we can backport to 2021.1

**Pull request merged**